src/edu/washington/escience/myria/operator/agg/AggUtils.java

```diff
@@ -84,14 +88,19 @@ public static boolean needsStats(final Set<AggregationOp> aggOps) {
    *
    * @param factories The factories that will produce the aggregators.
    * @param inputSchema The schema of the input tuples.
+   * @param pyFuncRegistrar - python function registrar to get functions from workers.
    * @return the aggregators for this operator.
    * @throws DbException if there is an error.
    */
   public static Aggregator[] allocateAggs(
-      final AggregatorFactory[] factories, final Schema inputSchema) throws DbException {
+      final AggregatorFactory[] factories,
+      final Schema inputSchema,
+      final PythonFunctionRegistrar pyFuncRegistrar)
```

Comment:
Again, I'm putting all the relevant comments here: I don't think it's necessary to extend all Aggregate-related constructors for passing `pyFuncRegistrar`. It's a per-worker property, and in some classes never used, which is a sign that putting it in the constructor is not a good design. Just get `pyFuncRegistrar` when we truly need it.